docs/memory_layout.md

```diff
@@ -0,0 +1,230 @@
+| Name           | Bytes                               |
+|----------------|-------------------------------------|
+| vec2\<float>[] | <kbd>[bbbb][bbbb][----][----]</kbd> |
+| vec3\<float>[] | <kbd>[bbbb][bbbb][bbbb][----]</kbd> |
+| vec4\<float>[] | <kbd>[bbbb][bbbb][bbbb][bbbb]</kbd> |
+| vec2\<int8>[]  | <kbd>[bb--][----][----][----]</kbd> |
+| vec2\<int32>[] | <kbd>[bbbb][bbbb][----][----]</kbd> |
+
+### STD430
+| Name           | Bytes                               |
+|----------------|-------------------------------------|
+| vec2\<float>[] | <kbd>[bbbb][bbbb]            </kbd> |
+| vec3\<float>[] | <kbd>[bbbb][bbbb][bbbb][----]</kbd> |
+| vec4\<float>[] | <kbd>[bbbb][bbbb][bbbb][bbbb]</kbd> |
+| vec2\<int8>[]  | <kbd>[bb]                    </kbd> |
+| vec2\<int32>[] | <kbd>[bbbb][bbbb]            </kbd> |
+| vec3\<int32>[] | <kbd>[bbbb][bbbb][bbbb][----]</kbd> |
+
+
+## Matrices (All matrices are column matrices)
+### STD140
+| Name           | Bytes                                                                                                                      |
+|----------------|----------------------------------------------------------------------------------------------------------------------------|
+| mat2x2\<int8>  | <kbd>[bb--][----][----][----]<br/>[bb--][----][----][----]                                                          </kbd> |
+| mat2x2\<fp16>  | <kbd>[bbbb][----][----][----]<br/>[bbbb][----][----][----]                                                          </kbd> |
+| mat2x2\<float> | <kbd>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]                                                          </kbd> |
+| mat2x3\<float> | <kbd>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]                                                          </kbd> |
+| mat2x4\<float> | <kbd>[bbbb][bbbb][bbbb][bbbb]<br/>[bbbb][bbbb][bbbb][bbbb]                                                          </kbd> |
+| mat3x2\<float> | <kbd>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]                             </kbd> |
+| mat4x2\<float> | <kbd>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]</kbd> |
+| mat4x3\<float> | <kbd>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]</kbd> |
+
+### STD430
+| Name           | Bytes                                                                                                                      |
+|----------------|----------------------------------------------------------------------------------------------------------------------------|
+| mat2x2\<fp16>  | <kbd>[bbbb][bbbb]                                                                                                   </kbd> |
+| mat2x2\<float> | <kbd>[bbbb][bbbb][bbbb][bbbb]                                                                                       </kbd> |
```

Comment:
Similar here.  Would write it as two rows.

Reply:
done